src/compiler/checker.ts

```diff
@@ -3080,7 +3080,7 @@ namespace ts {
         /**
          * Resolves a qualified name and any involved aliases.
          */
-        function resolveEntityName(name: EntityNameOrEntityNameExpression, meaning: SymbolFlags, ignoreErrors?: boolean, dontResolveAlias?: boolean, location?: Node): Symbol | undefined {
+        function resolveEntityName(name: EntityNameOrEntityNameExpression, meaning: SymbolFlags, ignoreErrors?: boolean, dontResolveAlias?: boolean, location?: Node, needSuggestedNameNotFoundMessage?: boolean): Symbol | undefined {

```

Comment:
what happens if you don't add `needSuggestedNameNotFoundMessage` and always request suggestions? It only makes a difference for `resolveName` calls that pass the combination `nameNotFoundMessage = undefined` and `issueSuggestions = true`. I'll go see if any calls like that exist.

Edit: I checked. The only new places that would issue suggestions is resolution for globals and JSX built-in symbols, only if they already request errors. This is fine. You can delete `needSuggestedNameNotFoundMessage` and `issueSuggestions` and unconditionally issue suggestions in `resolveName` if `message` is defined.